pkg/component/observability/monitoring/prometheus/cache/networkpolicy.go

```diff
@@ -45,3 +45,25 @@ func NetworkPolicyToNodeExporter(namespace string) *networkingv1.NetworkPolicy {
 		},
 	}
 }
+
+// NetworkPolicyToKubelet returns a NetworkPolicy that allows traffic from the
+// cache Prometheus to the kubelet process running on the nodes.
+func NetworkPolicyToKubelet(namespace string) *networkingv1.NetworkPolicy {
+	return &networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "egress-from-cache-prometheus-to-kubelet-tcp-10250",
+			Namespace: namespace,
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: monitoringutils.Labels(Label),
+			},
+			Egress: []networkingv1.NetworkPolicyEgressRule{{
+				To:    []networkingv1.NetworkPolicyPeer{},
```

Comment:
Do we omit the target IP range here as we do not have it in all cases? In theory, cache prometheus should only connect to worker nodes, i.e. the node CIDR should be a good target. However, I am not sure if we have that information, e.g. for unmanaged seeds.

Reply:
> cache prometheus should only connect to worker nodes, i.e. the node CIDR should be a good target

Thank you!

> I am not sure if we have that information, e.g. for unmanaged seeds.

I checked an unmanaged seed on the dev landscape, and the seed object contained the node CIDR range.
If it is not known under some circumstances, the restriction is not going to be applied in the current implementation.

I verified that the effect is as expected in the local setup: scraping is possible, but it is blocked if the IP range is not matching that of the nodes.

Please check if the approach is idiomatic: https://github.com/gardener/gardener/pull/9716/commits/c925e5bc6ad6c1a6bc2079363cc82944f04c8072

Reply:
Thanks for the change. It looks good and should definitively work in a calico environment. With cilium, there were problems using cluster internal IP ranges, but as far as I understood @axel7born this should now be resolved at least for the node range.

@axel7born Could you please check if this works for cilium with the current version or if there should additionally be the label selectors (of the `node-exporter` pods) added to the network policy?

Reply:
Do you know if cilium is supported in the KinD based local setup? [getting_started_locally.md
](https://github.com/gardener/gardener/blob/master/docs/deployment/getting_started_locally.md)
It is probably possible to create a shoot with cilium using the more involved "extensions" setup: [getting_started_locally_with_extensions.md](https://github.com/gardener/gardener/blob/master/docs/deployment/getting_started_locally_with_extensions.md). Due to the risk of leaking artifacts, I prefer to use the KinD based local setup, so testing this aspect in the extensions setup could be a bit challenging for me.

Reply:
@istvanballok I will test it later today with the extension setup. I'm not sure about the local setup and I wouldn't trust the result as networking behaves a bit different in the local setup.

Reply:
I just realized that we didn't set [policyCIDRMatchMode: nodes](https://docs.cilium.io/en/stable/security/policy/language/#selecting-nodes-with-cidr-ipblock)
With the current version of our cilium extension, I see dropped packets from cache-prometheus in Hubble.
We would need to merge https://github.com/gardener/gardener-extension-networking-cilium/pull/321 and build a new release.

Reply:
https://github.com/gardener/gardener-extension-networking-cilium/pull/321 has meanwhile been merged. We will probably create a new release soon.